app/uk/gov/hmrc/exports/services/NotificationService.scala

```diff
@@ -178,4 +194,24 @@ class NotificationService @Inject()(submissionRepository: SubmissionRepository,
     if (exportsPointer.isEmpty) logger.warn(s"Missing pointer mapping for [${wcoPointer}]")
     exportsPointer
   }
+
+  def reattemptParsingUnparsedNotifications(): Future[Unit] =
+    notificationRepository
+      .findUnparsedNotifications()
+      .map(_.foreach { unparsedNotification =>
+        try {
+          val notifications = parseNotificationsPayload(unparsedNotification.actionId, XML.loadString(unparsedNotification.payload))
+
+          //successfully parsed the previously unparsable notification
+          notifications.head.details.map { _ =>
+            saveAll(notifications)
```

Comment:
Can it possibly happen that there are no Notifications in the list? Calling `head` would result in exception then. Maybe it would be safer to use `headOption` and then flatten the `Option`s ?

Reply:
Fixed